internal/runner/operation_release.go

```diff
@@ -148,9 +151,27 @@ func (r *Runner) executeReleaseOp(
 			// Mark for deletion
 			ds = append(ds, d)
 		}
+		rl, err := r.client.ListReleases(ctx, &pb.ListReleasesRequest{
+			Application: app.Ref(),
+			Workspace:   project.WorkspaceRef(),
+		})
+		// Set up a slice of releases, whose max length is the # of releases
+		// we have
+		rs := make([]*pb.Release, 0, len(rl.Releases))
+		var rc int
+		for _, release := range rl.Releases {
+			for _, d := range ds {
+				if release.DeploymentId == d.Id {
+					rs = append(rs, release)
+					rc++
+				}
+			}
+
+		}
 
 		log.Info("will prune deploys", "len", len(ds))
 		pruneDeploys = ds
+		pruneReleases = rs[:rc]
```

Comment:
Why not set this to use `rs` directly instead? That would match how we build up the deployments slice too. Maybe also add a `log.Info` to match the prune deploy log.

Reply:
I did this because when I `make` the slice `rs`, it's the length of the total # of releases. Since the slice of releases we delete is a subset of our full list of releases, I set `pruneReleases` equal to the subset of the original slice, so that way we don't have empty elements in the `pruneReleases` slice.

The deployments list does something kind of similar, but further up in this file. However in this case, it is assumed that we're only pruning deployments older than 2 deployments ago. 
```go
// If we have less than the prune amount, then we do nothing. Otherwise
// we prune away the ones we're definitely keeping.
if len(resp.Deployments) <= retain {
	log.Debug("less than the limit deployments exists, no pruning")
	resp.Deployments = nil
} else {
	resp.Deployments = resp.Deployments[retain:]
}
```